docs/content/docs/ops/state/unaligned_checkpoints.md

```diff
@@ -0,0 +1,118 @@
+# Unaligned checkpoints
+
+Starting with Flink 1.11, checkpoints can be unaligned.
+[Unaligned checkpoints]({{< ref "docs/concepts/stateful-stream-processing" >}}#unaligned-checkpointing) 
+contain in-flight data (i.e., data stored in buffers) as part of the checkpoint state, allowing
+checkpoint barriers to overtake these buffers. Thus, the checkpoint duration becomes independent of
+the current throughput as checkpoint barriers are effectively not embedded into the stream of data
+anymore.
+
+You should use unaligned checkpoints if your checkpointing durations are very high due to
+backpressure. Then, checkpointing time becomes mostly independent of the end-to-end latency. Be
+aware unaligned checkpointing adds to I/O to the state backends, so you shouldn't use it when the
+I/O to the state backend is actually the bottleneck during checkpointing.
+
+### Alignment timeout
+
+After enabling unaligned checkpoints, you can also specify the alignment timeout programmatically:
+
+```java
+StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+env.getCheckpointConfig().setAlignmentTimeout(Duration.ofSeconds(30));
+```
+
+or in the `flink-conf.yml` configuration file:
+
+```
+execution.checkpointing.alignment-timeout: 30 s
+```
+
+When activated, each checkpoint will still begin as an aligned checkpoint, but if the alignment time
+for some subtask exceeds this timeout, then the checkpoint will proceed as an unaligned checkpoint.
+
+## Limitations
+
+### Concurrent checkpoints
+
+Flink currently does not support concurrent unaligned checkpoints. However, due to the more
+predictable and shorter checkpointing times, concurrent checkpoints might not be needed at all.
+However, savepoints can also not happen concurrently to unaligned checkpoints, so they will take
+slightly longer.
+
+### Interplay with watermarks
+
+Unaligned checkpoints break with an implicit guarantee in respect to watermarks during recovery.
+Currently, Flink generates the watermark as the first step of recovery instead of storing the latest
+watermark in the operators to ease rescaling. In unaligned checkpoints, that means on recovery,
+**Flink generates watermarks after it restores in-flight data**. If your pipeline uses an
+**operator that applies the latest watermark on each record** will produce **different results** than for
+aligned checkpoints. If your operator depends on the latest watermark being always available, the
+workaround is to store the watermark in the operator state. In that case, watermarks should be
+stored per key group in a union state to support rescaling.
+
+### Certain connections are not checkpointed
+
+There are types of connections with properties that are impossible to keep with channel data stored
+in checkpoints. To preserve these characteristics and ensure no state corruption or unexpected
+behaviour, unaligned checkpoints are disabled for such connections. All other exchanges still
+perform unaligned checkpoints.
```

Comment:
> All other exchanges still perform unaligned checkpoints.

replace with:
> Each and every one of other many remaining exhcanges types (keyBy) still perform unaligned checkpoints.

? ;)

Reply:
We should have a list what is working / what not.

Reply:
Isn't it unnecessarily wordy? I think it's better to give a clean explanation rather than to sound smart ;)

Reply:
If you meant to mention to add the `keyBy` I can do `All other exchanges (keyBy) still perform unaligned checkpoints.` if you like

Reply:
I was joking that `All other exchanges` is a bit strong statement if currently only single one is working.

Reply:
I see, I realized that only after my initial comment ;)

Reply:
@pnowojski , you do realize that we have ~10 partitioner and 2 are not working, right?